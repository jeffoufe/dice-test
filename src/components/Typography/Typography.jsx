// @flow

import React from 'react';
import styles from './Typography.module.scss';

import type { Node } from 'react';

type TypographyProps = {
    children: Node,
    className?: string,
    inline?: boolean
}

export const Text = ({ children, className, inline = false }: TypographyProps) => (
    <div className={`${styles.text} ${inline ? styles.inline : ''} ${className || ''}`}>{children}</div>
);

export const ImportantText = ({ children, className, inline = false }: TypographyProps) => (
    <div className={`${styles.important} ${inline ? styles.inline : ''} ${className || ''}}`}>{children}</div>
);

export const BlueHeader = ({ children, className, inline = false }: TypographyProps) => (
    <div className={`${styles['header-blue']} ${inline ? styles.inline : ''} ${className || ''}}`}>{children}</div>
);

export const GreyHeader = ({ children, className, inline = false }: TypographyProps) => (
    <div className={`${styles['header-grey']} ${inline ? styles.inline : ''} ${className || ''}}`}>{children}</div>
);