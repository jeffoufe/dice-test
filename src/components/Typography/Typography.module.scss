@import '../../constants/colors';

.text {
    font-size: 16px;
    line-height: 20px;
}

.inline {
    display: inline;
}

.important {
    font-size: 16px;
    line-height: 18px;
    font-weight: 600;
}

.blue-title {
    font-size: 14px;
    margin: 20px 0;
    line-height: 16px;
    text-transform: uppercase;
    font-weight: 600;
    color: $blue;
    letter-spacing: 0.05em;
}

.header {
    font-size: 14px;
    line-height: 16px;
    letter-spacing: 0.05em;
    font-weight: 600;
    margin-top: 20px;
    text-transform: uppercase;

    &-blue {
        @extend .header;
        color: $blue;
    }

    &-grey {
        @extend .header;
        color: $black;
        opacity: 0.5;
    }
}