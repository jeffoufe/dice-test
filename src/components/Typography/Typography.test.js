import { Text, BlueHeader, GreyHeader, ImportantText } from './Typography';
import { shallow } from 'enzyme';
import React from 'react';

describe('Featured', () => {
it('renders correctly', () => {
        expect(shallow(<Text>text</Text>)).toMatchSnapshot();
        expect(shallow(<BlueHeader inline>blue header</BlueHeader>)).toMatchSnapshot();
        expect(shallow(<GreyHeader>grey header</GreyHeader>)).toMatchSnapshot();
        expect(shallow(<ImportantText>important text</ImportantText>)).toMatchSnapshot();
    })
})