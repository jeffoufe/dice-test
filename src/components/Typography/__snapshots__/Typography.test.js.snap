// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured renders correctly 1`] = `
<div
  className="text  "
>
  text
</div>
`;

exports[`Featured renders correctly 2`] = `
<div
  className="header-blue inline }"
>
  blue header
</div>
`;

exports[`Featured renders correctly 3`] = `
<div
  className="header-grey  }"
>
  grey header
</div>
`;

exports[`Featured renders correctly 4`] = `
<div
  className="important  }"
>
  important text
</div>
`;
